components/lib/mention/Mention.js

```diff
@@ -94,9 +94,20 @@ export const Mention = React.memo(
             const { key, index } = triggerState;
             const value = inputRef.current.value;
             const position = DomHandler.getCursorOffset(inputRef.current, value.substring(0, index - 1), value.substring(index), key);
+            const viewportWidth = DomHandler.getViewport().width;
+            const inputRefClientRect = inputRef.current.getBoundingClientRect();
+            let left = position.left;
 
             overlayRef.current.style.transformOrigin = 'top';
-            overlayRef.current.style.left = `calc(${position.left}px + 1rem)`;
+
+            const isTouchingRight = left >= viewportWidth - inputRefClientRect.right;
+
+            if (isTouchingRight) {
+                overlayRef.current.style.left = `calc(${viewportWidth - inputRefClientRect.right}px - 1rem)`;
+            } else {
+                overlayRef.current.style.left = `calc(${position.left}px + 1rem)`;
+            }
```

Comment:
I wonder if its something that can be fixed in `DomHandler.flipfitCollision` more generically to be re-usable?

Reply:
@melloware  Do you think the below code should work ? 

```
        const alignOverlay = () => {
            DomHandler.flipfitCollision(inputRef.current, overlayRef.current, '', 'left bottom');
        };
```
right now its taking top:0, left:0 with above. Am i missing anything. Please let me know.

<img width="405" alt="Screenshot 2023-08-09 at 1 55 08 AM" src="https://github.com/primefaces/primereact/assets/38601497/d0352218-c33a-4404-98a7-2433cabf946c">

Reply:
I mean it should work but I tink it wasn't working when I tried it before?  Does it seem to be working now?

Reply:
flipfitCollision is working but its keeping the position top:0 and left:0 Let me look into this deeper tomorrow.